packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Models/Types/ModelTypeProviderTests.cs

```diff
@@ -110,5 +111,83 @@ public static IEnumerable<TestCaseData> BuildProperties_ValidatePropertySettersT
                     true);
             }
         }
+
+        private CSharpType GetCSharpType(InputType type)
+        {
+            switch (type)
+            {
+                case InputPrimitiveType:
+                    {
+                        var primitiveType = (InputPrimitiveType)type;
+
+                        if (primitiveType.Kind is InputTypeKind.String)
+                            return new CSharpType(typeof(string));
+
+                        if (primitiveType.Kind is InputTypeKind.Int32)
+                            return new CSharpType(typeof(int));
+
+                        throw new ArgumentException("Unsupported input type.");
+                    }
+                case InputList:
+                    return new CSharpType(typeof(IList<string>));
+                case InputDictionary:
+                    return new CSharpType(typeof(IDictionary<string, string>));
+                case InputIntrinsicType:
+                    return new CSharpType(typeof(BinaryData));
+                default:
+                    throw new ArgumentException("Unsupported input type.");
+            }
+        }
+
+        public void BuildConstructor_ValidateConstructors(InputModelTypeUsage usage)
```

Comment:
Where is this called?  if it is the root layer shouldn't it have a `[Test]` attribute?

Reply:
Yes, I'm not sure how it got removed. Perhaps I accidentally removed the [Test] attribute.